pkg/xray/resource_xray_workers_count.go

```diff
@@ -0,0 +1,280 @@
+package xray
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+func resourceXrayWorkersCount() *schema.Resource {
+	newContentSchema := map[string]*schema.Schema{
+		"new_content": {
+			Type:        schema.TypeInt,
+			Required:    true,
+			Description: "Number of workers for new content",
+		},
+	}
+
+	newExistingContentSchema := util.MergeMaps(
+		newContentSchema,
+		map[string]*schema.Schema{
+			"existing_content": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Number of workers for existing content",
+			},
+		},
+	)
+
+	workersCountSchema := map[string]*schema.Schema{
+		"index": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: "The number of workers managing indexing of artifacts.",
+			Elem: &schema.Resource{
+				Schema: newExistingContentSchema,
+			},
+		},
+		"persist": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: "The number of workers managing persistent storage needed to build the artifact relationship graph.",
+			Elem: &schema.Resource{
+				Schema: newExistingContentSchema,
+			},
+		},
+		"alert": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: "The number of workers managing alerts.",
+			Elem: &schema.Resource{
+				Schema: newExistingContentSchema,
+			},
+		},
+		"analysis": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: "The number of workers involved in scanning analysis.",
+			Elem: &schema.Resource{
+				Schema: newExistingContentSchema,
+			},
+		},
+		"impact_analysis": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: "The number of workers involved in Impact Analysis to determine how a component with a reported issue impacts others in the system.",
+			Elem: &schema.Resource{
+				Schema: newContentSchema,
+			},
+		},
+		"notification": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: "The number of workers managing notifications.",
+			Elem: &schema.Resource{
+				Schema: newContentSchema,
+			},
+		},
+	}
+
+	type NewContent struct {
+		New int `json:"new_content"`
```

Comment:
So: 
```go 
   thing := NewContent{ 
     New : 10
    }
   thing.New = 4
```
a field called `New` would confuse me since you're not using it's name to derive it's json. Does it have to be called `New`?